source/jormungandr/jormungandr/scenarios/journey_filter.py

```diff
@@ -366,28 +366,37 @@ def is_walk_after_parking(journey, idx_section):
     return False
 
 
-def similar_journeys_vj_generator(journey):
+def similar_pt_section_vj(section):
+    return 'pt:%s' % section.pt_display_informations.uris.vehicle_journey
+
+
+def similar_pt_section_line(section):
+    return "pt:{}".format(section.pt_display_informations.uris.line)
+
+
+def similar_journeys_generator(journey, pt_functor):
     for idx, s in enumerate(journey.sections):
         if s.type == response_pb2.PUBLIC_TRANSPORT:
-            yield 'pt:%s' % s.pt_display_informations.uris.vehicle_journey
+            yield pt_functor(s)
         elif s.type == response_pb2.STREET_NETWORK:
             # special case, we don't want to consider the walking section after/before parking a car
             # so CAR / PARK / WALK / PT is equivalent to CAR / PARK / PT
             if is_walk_after_parking(journey, idx):
                 continue
             yield 'sn:%s' % s.street_network.mode
+        elif s.type == response_pb2.CROW_FLY and s.street_network.mode == response_pb2.Ridesharing:
```

Comment:
Why not distinguish any different crowfly mode?
Maybe also yield that it's a crowfly to distinguish it from a classic street-network.
I'm think of SNCF's coverage, we'd want to keep bike, car and walk if we have them in params, no?

Reply:
I didn't want to alter what is done here, I have only handled crowfly since it's very different of others modes as it will be filtered later and can remove journeys from the final response.
As for distinguish crowfly from sn I don't really have any preferences, I don't see a case where we will have both that use the same vj.

Reply:
OK, I think we should do it for all the modes.
As for the crowfly vs. SN no opinion also, it would be a guard but I don't imagine any case where it would happen.

Reply:
@pbougue: I'm not sure I get the benefits from distinguishing crowflies for all different modes ? Maybe I'm missing something here.

If we can't think of any use case and we don't have any need for this, I suggest to keep it this way.

Reply:
If we ask for first_section_mode=[bike, walk] on SNCF and hop on the same train in both cases, one is filtered if we don't distinguish it, as it's the case currently : https://api.navitia.io/v1/coverage/sncf/journeys?from=-1.53564%3B47.22167&to=stop_area%3AOCE%3ASA%3A87391003&datetime=20181107T065000&first_section_mode%5B%5D=bike&first_section_mode%5B%5D=walking&
In my opinion we want both in that case.

Reply:
huuummmmm interesting! I agree that it would make sense to differentiate the 2 types of crowfly (the same way we would do with street network).